db/data.py

```diff
@@ -187,6 +189,22 @@ def _insert_lowlevel_json(connection, ll_id, data_json, data_sha256, version_id)
                                    "data_sha256": data_sha256,
                                    "version": version_id})
 
+    def _get_submission_offset(connection, mbid):
+        """ Get highest existing submission offset for mbid, then increment """
+        query = text("""
+            SELECT MAX(submission_offset) as max_offset
+              FROM lowlevel
+             WHERE gid = :mbid
```

Comment:
@alastair I've removed the cast `gid::text` here, which allowed the query to work without a call to `str()` in `test_stats` file.